rust/src/builder.rs

```diff
@@ -57,21 +144,34 @@ where
 
     /// Returns the current capacity of the builder (number of elements)
     pub fn capacity(&self) -> i64 {
-        let byte_capacity = self.buffer.capacity();
-        (byte_capacity / mem::size_of::<T>()) as i64
+        let byte_capacity = self.buffer.capacity() as i64;
+        byte_capacity * 8
     }
 
     /// Push a value into the builder, growing the internal buffer as needed
-    pub fn push(&mut self, v: T) -> Result<()> {
+    pub fn push(&mut self, v: bool) -> Result<()> {
         self.reserve(1)?;
-        self.write_bytes(v.to_byte_slice(), 1)
+        if v {
+            // Check needed here to increase the buffers `len` to ensure that `set_bit` is safe
+            // Zeroing out other bits also makes testing easier
+            if self.len > self.buffer.len() as i64 * 8 {
```

Comment:
Should this be `>=`? Also, it may be too expensive to do this check for every bit we write. Can we zero out the bits when we grow the capacity so that we only need to do it once?

Reply:
Yep it should be `>=`, thanks